src/overlays/actors/ovl_En_Trt2/z_en_trt2.c

```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
+    return RADF_TO_BINANG(Math_Acot2F(phi_f12, phi_f14));
+}
+
+f32 func_80AD49B8(Path* path, s32 arg1, Vec3f* arg2, Vec3s* arg3) {
+    s32 pad;
+    Vec3f sp20;
+    Vec3s* temp_v1;
+
+    if (path != NULL) {
+        temp_v1 = Lib_SegmentedToVirtual(path->points);
+        temp_v1 = &temp_v1[arg1];
+
+        sp20.x = temp_v1->x;
+        sp20.y = temp_v1->y;
+        sp20.z = temp_v1->z;
+    }
+    arg3->y = Math_Vec3f_Yaw(arg2, &sp20);
+    arg3->x = Math_Vec3f_Pitch(arg2, &sp20);
+    return sp20.y - arg2->y;
+}
+
+void func_80AD4A78(EnTrt2* this, GlobalContext* globalCtx) {
+    Vec3f sp34;
+
+    sp34.x = randPlusMinusPoint5Scaled(15.0f) + this->actor.world.pos.x;
+    sp34.y = this->actor.world.pos.y;
+    sp34.z = randPlusMinusPoint5Scaled(15.0f) + this->actor.world.pos.z;
+    Actor_SpawnFloorDustRing(globalCtx, &this->actor, &sp34, 50.0f, 0, 2.0f, 0, 0, 0);
+}
+
+s32 func_80AD4B08(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 ret;
+
+    if (SurfaceType_GetSceneExitIndex(&globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId) == 0x15) {
+        ret = true;
+    } else {
+        ret = false;
+    }
+    return ret;
+}
+
+s32 func_80AD4B4C(EnTrt2* this, GlobalContext* globalCtx) {
+    s32 sp24 = false;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        sp24 = true;
+        this->actor.speedXZ = 0.0f;
+        func_80AD349C(this);
+        this->unk_3B4 = this->unk_3B2;
+        if (this->actor.bgCheckFlags & 1) {
+            if ((player->transformation != PLAYER_FORM_HUMAN) && (player->transformation != PLAYER_FORM_FIERCE_DEITY)) {
+                this->unk_3A8 = 0x84F;
+            }
+            func_80AD3DA4(this, globalCtx);
+        } else if ((player->transformation == PLAYER_FORM_HUMAN) ||
+                   (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+            this->unk_3B2 = 16;
+            this->actor.velocity.y = 1.5f;
+        } else {
+            this->unk_3A8 = 0x84F;
+            this->unk_3D8 = true;
+            func_80AD3380(&this->skelAnime, sAnimations, 7);
+            this->unk_3B2 = 7;
+        }
+    }
+    return sp24;
+}
+
+s32 func_80AD4C4C(EnTrt2* this) {
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            return true;
+        }
+    } else if ((this->actor.xzDistToPlayer > 50.0f) && (this->actor.xzDistToPlayer < 200.0f)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80AD4CCC(EnTrt2* this, GlobalContext* globalCtx) {
+    s16 sp1E = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (((this->unk_3B2 == 4) || (this->unk_3B2 == 5)) && this->actor.isTargeted && !(this->actor.bgCheckFlags & 1) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
```

Comment:
```suggestion
    if ((this->unk_3B2 == 4 || this->unk_3B2 == 5) && this->actor.isTargeted && 
        !(this->actor.bgCheckFlags & BGCHECK_CHECK_WALL) && 
        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
```
formatter probably still wont like that